presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1398,15 +1414,15 @@ private static ExpressionAnalysis analyzeExpressions(
         }
 
         return new ExpressionAnalysis(
-                analyzer.getExpressionTypes(),
-                analyzer.getExpressionCoercions(),
-                analyzer.getSubqueryInPredicates(),
-                analyzer.getScalarSubqueries(),
-                analyzer.getExistsSubqueries(),
-                analyzer.getColumnReferences(),
-                analyzer.getTypeOnlyCoercions(),
-                analyzer.getQuantifiedComparisons(),
-                analyzer.getLambdaArgumentReferences());
+                NodeRefCollections.toIdentityMap(analyzer.getExpressionTypes()),
```

Comment:
is it really required to convert from/to `IdentityLinkedHashMap` back and forth? Why not just expose `Map<NodeRef<...>, T>`?